### PR TITLE
Fixed #29282 -- Made admin checks handle TypeError

### PR DESCRIPTION
**Pull request by dsanders11**
https://code.djangoproject.com/ticket/29282

### CODE REVIEW

**Comment by dsanders11** [comment]
Build failure seems unrelated.

File: tests/modeladmin/test_checks.py [hunk 1]
```diff
@@ -639,6 +639,18 @@ class TestModelAdmin(ModelAdmin):
             'admin.E115'
         )
 
+    def test_not_filter_again_again_again(self):
```
Comment (timgraham) [inline tests/modeladmin/test_checks.py]: I'd suggest a name like `test_list_filter_is_function`.

File: tests/modeladmin/test_checks.py [hunk 2]
```diff
@@ -639,6 +639,18 @@ class TestModelAdmin(ModelAdmin):
...
+        def get_filter():
+            pass
+
+        class TestModelAdmin(ModelAdmin):
+            list_filter = (get_filter,)
+
+        self.assertIsInvalid(
+            TestModelAdmin, ValidationTestModel,
+            "The value of 'list_filter[0]' must inherit from 'ListFilter'.",
+            'admin.E113')
```
Comment (timgraham) [inline tests/modeladmin/test_checks.py]: Please put the closing ) on the next line as in other tests.

**Comment by akulakov** [comment]
Alternative names:
issubclass_skip_raise_exception
issubclass_do_not_raise
issublcass_no_exception

I prefer @timgraham suggestion to current name, as well.

File: tests/modeladmin/test_checks.py (hunk 2 above)
Comment (dsanders11) [reply tests/modeladmin/test_checks.py]: This was a side-effect of copy-and-pasting from line 604-607 which has it on the same line. Do you want me to fix that one as well for consistency?

**Comment by dsanders11** [comment]
I'm fine with any of the suggestions @akulakov made, with my choice being `issubclass_do_not_raise`. I agree with Tim that `issubclass_` may not be a better name.

File: tests/modeladmin/test_checks.py (hunk 2 above)
Comment (timgraham) [reply tests/modeladmin/test_checks.py]: No, not necessary.

**Comment by timgraham** [comment]
I pushed a small update, however, complete test coverage and more code changes are needed. For example, in `_check_inlines_Item()`, `issubclass()` won't be reached because `inline_label = inline.__module__ + '.' + inline.__name__` will crash first. Do you want to complete this David? If not, I'll mark the ticket as "easy pickings" since the remaining changes shouldn't be too difficult.

**Comment by CuriousLearner** [comment]
Hi @dsanders11 , Are you working on this bug? If not, then I'd like to help out get this in. :)

cc @timgraham

**Comment by timgraham** [comment]
Closing due to inactivity.

**Pull request closed without merging**